tests/unit/test_service.py

```diff
@@ -813,6 +813,67 @@ def test_resource_remove(self):
         self.exporter.strategy.remove.accept_called()
 
 
+class TestDCGMSnapExporter(unittest.TestCase):
+    """Test DCGM Snap exporter's methods."""
+
+    def setUp(self) -> None:
+        """Set up harness for each test case."""
+        snap_lib_patcher = mock.patch.object(service, "snap")
+        shutil_lib_patcher = mock.patch.object(service, "shutil")
+        self.mock_snap = snap_lib_patcher.start()
+        self.mock_shutil = shutil_lib_patcher.start()
+        self.addCleanup(snap_lib_patcher.stop)
+        self.addCleanup(shutil_lib_patcher.stop)
+
+        search_path = pathlib.Path(f"{__file__}/../../..").resolve()
+        self.mock_config = {
+            "dcgm-snap-channel": "latest/stable",
+        }
+
+        self.exporter = service.DCGMExporter(search_path, self.mock_config)
+        self.exporter.strategy = mock.MagicMock()
+
+    def test_exporter_name(self):
+        self.assertEqual(self.exporter.exporter_name, "dcgm")
+
+    def test_hw_tools(self):
+        self.assertEqual(self.exporter.hw_tools(), {HWTool.DCGM})
+
+    def test_install_failed(self):
+        self.exporter.snap_client.present = False
+
+        exporter_install_ok = self.exporter.install()
+
+        self.exporter.strategy.install.accept_called()
+        self.mock_shutil.copy.accept_not_called()
```

Comment:
```suggestion
        self.exporter.strategy.install.assert_called()
        self.mock_shutil.copy.assert_not_called()
```

It looks like these have been widely used typos in this codebase.  Called typoed methods here don't fail because they're mocks.

Please update accept_called and accept_not_called to assert_called and assert_not_called in all the code you've added here - it's out of scope for the rest of the codebase in this PR.

A hint for testing tests is do something to purposefully make the test fail, run the test, and verify it fails (eg. for here it could be commenting out the strategy.install() call and checking this test fails, or inverting the condition in the test and checking it fails).

Reply:
Opened a bug report for fixing the other cases https://github.com/canonical/hardware-observer-operator/issues/329

Reply:
Thanks for pointing that out!